xdsl/dialects/x86.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import ClassVar
+
+from typing_extensions import Self
+
+from xdsl.ir import (
+    Data,
+    Dialect,
+    TypeAttribute,
+)
+from xdsl.irdl import (
+    irdl_attr_definition,
+)
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
+from xdsl.utils.exceptions import VerifyException
+
+
+class X86RegisterType(Data[str], TypeAttribute, ABC):
+    """
+    An x86 register type.
+    """
+
+    _unallocated: ClassVar[Self | None] = None
+
+    @classmethod
+    def unallocated(cls) -> Self:
+        if cls._unallocated is None:
+            cls._unallocated = cls("")
+        return cls._unallocated
```

Comment:
We moved away from this pattern in riscv, `UNALLOCATED` should probably be in the `Registers` as one of the options, and the `unallocated` class method abstract

Reply:
It wouldn't because I'm only suggesting to move the abstract base class to the backend module, which has mostly abstract methods. The riscv and x86-specific subclasses should remain with their dialects

Reply:
Ah yes

Reply:
Maybe we'll need to refactor that then

Reply:
sure, sounds good